tests/tb_gateway_mqtt_client_tests.py

```diff
@@ -13,10 +13,287 @@
 # limitations under the License.
 
 import unittest
+from unittest.mock import MagicMock,patch
 from time import sleep, time
+import threading
+from tb_gateway_mqtt import TBGatewayMqttClient, TBSendMethod, GATEWAY_CLAIMING_TOPIC, GATEWAY_RPC_TOPIC, GATEWAY_MAIN_TOPIC
 
-from tb_gateway_mqtt import TBGatewayMqttClient
 
+class TestGwUnsubscribe(unittest.TestCase):
+    def setUp(self):
+        self.client = TBGatewayMqttClient("localhost", 1883, "dummy_token")
+        if not hasattr(self.client, "_lock"):
```

Comment:
What this check for? As mentioned in another comment - this lock creates during initialisation of TBDeviceMqttClient instance.